File: 2_Introduction to C++ for DSA/8_Practice_Problem_6.5/I_Palindrome.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
    string s;
    cin >> s;

    int i = 0, j = s.size() - 1;
    int flag = 1;

    while (i < j)
    {
        if(s[i] != s[j])
        {
            flag = 0;
            break;
        }
        i++;
        j--;
    }

    flag == 1 ? cout << "YES" : cout << "NO";

    return 0;
}

File: 6_SDT/Week 01- Introduction to Python/Module 02.5-Practice day 01/I_Palindrome.py
num = str(input())
rev = num[::-1]

print(int(rev))

if num == rev:
    print("YES")
else:
    print("NO")
        
        
        
        
"""
string s; cin >> s;
string cpy = s;
reverse(cpy.begin(), cpy.end());

cout << stoll(rev) << nl;
if(s == cpy)
    cout << "YES";
else cout << "NO";

"""